staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -683,11 +686,14 @@ func (s *store) watch(ctx context.Context, key string, rv string, pred storage.S
 	return s.watcher.Watch(ctx, key, int64(rev), recursive, pred)
 }
 
-func (s *store) getState(getResp *clientv3.GetResponse, key string, v reflect.Value, ignoreNotFound bool) (*objState, error) {
+func (s *store) getState(getResp *clientv3.GetResponse, key string, v reflect.Value, hubGroupVersionKind schema.GroupVersionKind, ignoreNotFound bool) (*objState, error) {
 	state := &objState{
 		obj:  reflect.New(v.Type()).Interface().(runtime.Object),
 		meta: &storage.ResponseMeta{},
 	}
+	// Set the hubGroupVersionKind into the new instance
+	// This is needed by unstructured decoders to know what version we want and convert appropriately
+	state.obj.GetObjectKind().SetGroupVersionKind(hubGroupVersionKind)
```

Comment:
this feels like bandaid. It works, but the real issue is that we create state.obj above via reflection. Instead this has to go through the strategy's New method.

Reply:
plumbing the New() method from the strategy required changing Storage interfaces, which I really don't want to do. Reworked this to make the `copy+set GVK` behavior part of the Unstructured interface and checked for that here